components/service/pocket/src/main/java/mozilla/components/service/pocket/net/PocketResponse.kt

```diff
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.pocket.net
+
+/**
+ * A response from the Pocket API: the subclasses determine the type of the result and contain usable data.
+ */
+sealed class PocketResponse<T> {
+
+    /**
+     * A successful response from the Pocket API.
+     *
+     * @property data The data returned from the Pocket API.
+     */
+    data class Success<T> internal constructor(val data: T) : PocketResponse<T>()
+
+    /**
+     * A failure response from the Pocket API.
+     */
+    class Failure<T> internal constructor() : PocketResponse<T>()
+}
```

Comment:
This is actually a very generic class and I wonder if we would want such a class in the base component.

Reply:
@pocmo It's up to you – if so, let me know! fwiw, Severin suggested returning a wrapping sealed class, rather than null or an empty list directory because:
- RxJava does not support null values so the conusmer would have to wrap anyway
- Returning a sealed class lets us extend the API to add more information later without breaking changes (e.g. adding the exception to the failure).

Reply:
Alright. Something to consider later on. I guess in Arrow that would be a [Try](https://arrow-kt.io/docs/arrow/core/try/).

Reply:
@pocmo Woah woah woah: this is in the kotlin stdlib! `Result`: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/index.html

I'll open a PR to change the code over to Result.